### PR TITLE
Add documentation with Documenter.jl

### LINKED ISSUE #936: Documentation?
Given that this is one of the most popular Julia packages, I think it would be very useful to have some actual documentation generated by `Documenter.jl`. For one, it would be very helpful to have some more documentation about the API and the internals (these might be present in docstrings, but you have to go searching for them). I think this creates a barrier to those like myself who would otherwise be interested in contributing to the project.

If the maintainers are open to this I can definitely start putting together a documentation skeleton to get the ball rolling.

### PR DESCRIPTION
**Pull request by jlumpe**
Created `make.jl` and documentation source files with "Manual" section containing contents of README. Also added "Library" section with a "Public API" page containing `@audodoc` for module and an "Internals" page which is currently blank.

Closes #936.

### CODE REVIEW

**Comment by stevengj** [comment]
Presumably, this information should be removed from the README and replaced with a link to the docs…

**Comment by jlumpe** [comment]
@stevengj Sure, what exactly should be left in it?

**Comment by stevengj** [comment]
Probably just the first paragraph, plus a quick-start that suggests
```
using IJulia
notebook()
```

**Comment by jlumpe** [comment]
@stevengj done. The documentation badge can probably be added once the hosting is set up?

**Comment by stevengj** [comment]
We'll need to add:
```
after_success:
  - julia -e 'cd(Pkg.dir("QuadGK")); ps=Pkg.PackageSpec(name="Documenter", version="0.19"); Pkg.add(ps); Pkg.pin(ps); include(joinpath("docs", "make.jl"))';
```
in `.travis.yml`, and
```jl
deploydocs(
    julia = "nightly",
    repo = "github.com/JuliaLang/IJulia.jl.git",
    target = "build",
    deps = nothing,
    make = nothing,
)
```
to `make.jl`, and 
```
Documentation:
[![](https://img.shields.io/badge/docs-stable-blue.svg)](https://JuliaLang.github.io/IJulia.jl/stable)
[![](https://img.shields.io/badge/docs-latest-blue.svg)](https://JuliaLang.github.io/IJulia.jl/latest)
```
to the README

**Comment by jlumpe** [comment]
Tried it with the GitHub workflow instead of Travis CI because I wasn't too sure how the authentication worked. More or less copied from https://juliadocs.github.io/Documenter.jl/stable/man/hosting. I'm not really familiar with GH workflows either but it seems like it ran successfully for this PR. I think you should be able to merge this and then enable Github pages for this repo?

File: .github/workflows/documentation-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@latest
+        with:
+          version: '1.4'
+      - name: Install dependencies
+        run: julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.instantiate(); Pkg.build("IJulia")'
+      - name: Build and deploy
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # For authentication with GitHub Actions token
```
Comment (stevengj) [inline .github/workflows/documentation-workflow.yml:22]: Seems like we will run into https://github.com/JuliaDocs/Documenter.jl/issues/1177?

Comment (jlumpe) [reply .github/workflows/documentation-workflow.yml:22]: Seems like the problem is with initially creating the `gh-pages` branch, the following comment says you can get around it by making an initial (empty) commit to it:
https://github.com/JuliaDocs/Documenter.jl/issues/1177#issuecomment-640998242

Comment (stevengj) [reply .github/workflows/documentation-workflow.yml:22]: Okay, I created an empty `gh-pages` branch.

**Comment by stevengj** [comment]
Hmm, Github pages is enabled and set to use the `gh-pages` branch, but I'm still getting a broken link for the docs.   Is there something else I need to do?
image

**Comment by jlumpe** [comment]
I'm able to visit the hosted documentation by clicking the badges added to the README, it looks fine to me.

**Comment by stevengj** [comment]
Yes, @DilumAluthge helped me to fix it.

**Pull request merged**